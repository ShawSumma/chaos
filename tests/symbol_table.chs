bool var1 = true
boolean var2 = false
bool var3 = true
bool var4 = false
var1 = false
var2 = true
del var2
num var5 = 5
var5 = 7
number var6 = 3.2
var6 = 5.4
str var7 = 'string1'
var7 = 'string2'
string var8 = 'string3'
var8 = 'string4'
symbol_table
