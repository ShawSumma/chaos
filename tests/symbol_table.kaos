bool var1 = true
bool var2 = false
bool var3 = true
bool var4 = false
var1 = false
var2 = true
del var2
num var5 = 5
var5 = 7
num var6 = 3.2
var6 = 5.4
str var7 = 'string1'
var7 = 'string2'
str var8 = 'string3'
var8 = 'string4'
symbol_table
