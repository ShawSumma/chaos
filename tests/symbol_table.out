[start] =>
	{name: var1, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 0, 2nd_type: 0, value_type: 0, role: 0, param_of: } =>
	{name: var3, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 0, 2nd_type: 0, value_type: 0, role: 0, param_of: } =>
	{name: var4, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 0, 2nd_type: 0, value_type: 0, role: 0, param_of: } =>
	{name: var5, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 2, role: 0, param_of: } =>
	{name: var6, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 2, role: 0, param_of: } =>
	{name: var7, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
	{name: var8, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 2, 2nd_type: 0, value_type: 3, role: 0, param_of: } =>
[end]
